### PR TITLE
Update Koans

### PR DESCRIPTION
**Pull request by JonathanStarup**
Solving #2153

**Main changes:**
* `def` -> `pub def`
* datalog input/ouput is changed to arrays instead
* unshorten names
* projecting and queries requre `Boxable[city]` fx. so that is added

### CODE REVIEW

File: examples/koans/drivable-speed.flix [hunk 1]
```diff
@@ -0,0 +1,7 @@
+pub def drivable(roads: Array[(city, Int, city)], source: city, destination: city, minimumSpeed: Int): Bool  with Boxable[city] =
```
Comment (magnus-madsen) [inline examples/koans/drivable-speed.flix:1]: I am surprised this function is able to be typed as pure! How?

**Comment by magnus-madsen** [comment]
I note that you use polymorphic types everywhere, e.g. `city. I assume thats on purpose? I guess otherwise everything would have to be strings. Would that be better or?

**Comment by magnus-madsen** [comment]
Once this is merged in, flix.dev should be updated.

**Comment by JonathanStarup** [comment]
> I note that you use polymorphic types everywhere, e.g. `city. I assume thats on purpose? I guess otherwise everything would have to be strings. Would that be better or?

Yes, I think its nice and makes it almost paste-able into other code. It does add the `Boxed[type]` though

File: examples/koans/drivable-speed.flix (hunk 1 above)
Comment (JonathanStarup) [reply examples/koans/drivable-speed.flix:1]: I was thinking about that too. Creating an array is impure right? but querying and projecting is pure even though the operate on arrays. Maybe this should be changed? It is nice that queries are pure though but to actually be pure it should probably be lists all around.

**Comment by magnus-madsen** [comment]
Merged! Thanks. Will you update the flix.dev repository too? Just make a PR with these changes.

**Pull request merged**